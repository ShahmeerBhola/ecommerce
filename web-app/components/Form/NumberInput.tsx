import React from 'react';
import InputNumber from 'rc-input-number';
import { InputNumberProps } from 'rc-input-number/lib/interface';

import './number_input.scss';

type Props = Omit<InputNumberProps, 'style'>;

const NumberInput: React.FunctionComponent<Props> = ({ min = 0, ...props }) => (
  <InputNumber
    min={min}
    style={{ display: 'flex', flexDirection: 'row-reverse', alignItems: 'center', justifyContent: 'space-between' }}
    focusOnUpDown
    {...props}
  />
);

export default NumberInput;
