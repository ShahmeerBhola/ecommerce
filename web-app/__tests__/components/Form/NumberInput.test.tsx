import React from 'react';
import renderer from 'react-test-renderer';
import NumberInput from '../../../components/Form/NumberInput';

describe('NumberInput component', () => {
  test('it renders properly', () => {
    const tree = renderer.create(<NumberInput value={5} onChange={(): void => console.log('Hello!')} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
