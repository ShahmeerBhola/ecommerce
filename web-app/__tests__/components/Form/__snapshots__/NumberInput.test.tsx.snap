// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput component it renders properly 1`] = `
<input
  min={1}
  onChange={[Function]}
  type="number"
  value={5}
/>
`;
